### ISSUE TITLE
[Research] Add EmoAdapt Self-Supervised Feature Learning for 4-Channel EEG (+20% over heuristics)

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#29]
## What

Implement **EmoAdapt** (MDPI Mathematics, May 2025) — a self-supervised feature extractor purpose-built for **4-channel consumer EEG** emotion recognition. Combines contrastive learning + masked prediction to learn robust emotional features from limited channels. Achieves **+20.4% improvement** over conventional feature-based approaches.

## Why

- This is the **most directly applicable paper** to NeuralDreamWorkshop's hardware setup
- Built and tested on 4-channel consumer EEG (same as Muse 2)
- Current system: hand-crafted features (band powers, ratios, FAA) → 60-74%
- EmoAdapt: self-supervised features from same hardware → **+20.4% accuracy**
- Handles low SNR and limited channels inherent to consumer EEG
- No labeled data needed for feature learning (self-supervised pretraining)
- Cross-session accuracy: 60.2% (better than most cross-subject results)

### Step 1: Implement EmoAdapt feature extractor
```python
# ml/models/emoadapt.py
class EmoAdapt(nn.Module):
    """Self-supervised feature extractor for 4-channel consumer EEG.
    
    Pre-training objectives:
    1. Contrastive: augmented views of same EEG epoch should be close
    2. Masked prediction: predict masked time/channel segments
    
    Input: (4, 1024) = 4 channels × 4 seconds at 256 Hz
    Output: 128-dim feature vector (replaces hand-crafted 41 features)
    """
    def __init__(self):
        self.encoder = nn.Sequential(
            # Temporal convolutions per channel
            # Cross-channel attention
            # Projection head (for contrastive loss)
        )
    
    def pretrain(self, unlabeled_eeg_data):
        """Self-supervised pretraining on unlabeled Muse 2 data.
        No emotion labels needed — learns from signal structure.
        """
    
    def extract_features(self, eeg_epoch):
        """Extract learned features for downstream classification."""
        return self.encoder(eeg_epoch)  # 128-dim vector
```

### Step 2: Self-supervised pretraining
- Collect unlabeled EEG data from Muse 2 sessions (no labels needed!)
- All existing session data can be used for pretraining
- Augmentations: time shifting, channel dropout, noise injection, amplitude scaling
- Train for ~50-100 epochs

### Step 3: Fine-tune for emotion classification
- After pretraining: freeze encoder, train linear classifier on labeled data
- Or: use EmoAdapt features + existing LightGBM classifier
- Compare: hand-crafted 41 features vs EmoAdapt 128-dim features

### Step 4: Integrate into inference chain
- Add as feature extraction option in `emotion_classifier.py`
- Can be combined with existing features: [41 hand-crafted + 128 learned = 169 features]

### Files to Modify
- `ml/models/emoadapt.py` (NEW) — Self-supervised feature extractor
- `ml/training/pretrain_emoadapt.py` (NEW) — Pretraining script
- `ml/models/emotion_classifier.py` — Add EmoAdapt feature path
- `ml/processing/eeg_processor.py` — Add EmoAdapt feature extraction function

### Expected Impact
- Feature quality: **+20.4%** over hand-crafted features (paper's result)
- Cross-session emotion accuracy: **60-74% → 75-85%**
- Uses ALL existing unlabeled session data (no labeling effort)

## Complexity
MEDIUM — standard PyTorch contrastive learning, no external pretrained weights needed

## References
- EmoAdapt paper: https://www.mdpi.com/2227-7390/13/10/1608
- "A Portable and Affordable Four-Channel EEG System for Emotion Recognition with Self-Supervised Feature Learning"
- 4-channel consumer EEG, dry electrodes, wireless — matches Muse 2
